pkg/kubefed2/unjoin.go

```diff
@@ -281,81 +230,90 @@ func unregisterCluster(crClientset *crclient.Clientset, clusterNamespace, host,
+	} else {
+		glog.V(2).Infof("Deleted FederatedCluster resource from namespace: %s for unjoin cluster: %s", federationNamespace, unjoiningClusterName)
 	}
-
-	return nil, nil
 }
 
 // deleteRBACResources deletes the cluster role, cluster rolebindings and service account
 // from the unjoining cluster.
-func deleteRBACResources(hostClusterClientset, unjoiningClusterClientset client.Interface,
-	namespace, unjoiningClusterName, hostClusterName string, dryRun bool) error {
+func deleteRBACResources(unjoiningClusterClientset client.Interface,
+	namespace, unjoiningClusterName, hostClusterName string, dryRun bool) bool {
+
+	var isSucceed = true
 
 	saName := util.ClusterServiceAccountName(unjoiningClusterName, hostClusterName)
 
 	glog.V(2).Infof("Deleting cluster role binding for service account: %s in unjoining cluster: %s",
 		saName, unjoiningClusterName)
 
-	err := deleteClusterRoleAndBinding(unjoiningClusterClientset, saName, namespace, dryRun)
-	if err != nil {
-		glog.V(2).Infof("Error deleting cluster role binding for service account: %s in unjoining cluster: %v",
-			saName, err)
-		return err
+	isSucceed = deleteClusterRoleAndBinding(unjoiningClusterClientset, saName, namespace, dryRun)
+	if isSucceed {
+		glog.V(2).Infof("Deleted cluster role binding for service account: %s in unjoining cluster: %s",
+			saName, unjoiningClusterName)
 	}
 
-	glog.V(2).Infof("Deleted cluster role binding for service account: %s in unjoining cluster: %s",
-		saName, unjoiningClusterName)
-
 	glog.V(2).Infof("Deleting service account %s in unjoining cluster: %s", saName, unjoiningClusterName)
 
-	err = deleteServiceAccount(unjoiningClusterClientset, saName, namespace, dryRun)
+	err := deleteServiceAccount(unjoiningClusterClientset, saName, namespace, dryRun)
 	if err != nil {
-		return err
+		isSucceed = false
+		glog.V(2).Infof("Error deleting service account: %s in unjoining cluster. %v", saName, err)
+	} else {
+		glog.V(2).Infof("Deleted service account %s in unjoining cluster: %s", saName, unjoiningClusterName)
 	}
 
-	glog.V(2).Infof("Deleted service account %s in unjoining cluster: %s", saName, unjoiningClusterName)
-
-	return nil
+	return isSucceed
 }
 
 // deleteFedNSFromUnjoinCluster deletes the federation namespace and multi cluster
 // public namespace in the unjoining cluster.
 func deleteFedNSFromUnjoinCluster(unjoiningClusterClientset client.Interface,
-	federationNamespace, unjoiningClusterName, hostClusterName string, dryRun bool) error {
+	federationNamespace, unjoiningClusterName string, dryRun bool) error {
 
 	if dryRun {
 		return nil
 	}
 
+	glog.V(2).Infof("Deleting federation namespace: %s from unjoin cluster: %s",
+		federationNamespace, unjoiningClusterName)
 	err := unjoiningClusterClientset.CoreV1().Namespaces().Delete(federationNamespace,
 		&metav1.DeleteOptions{})
 	if err != nil {
 		glog.V(2).Infof("Could not delete federation namespace: %s due to: %v", federationNamespace, err)
```

Comment:
If the strategy of returning an error is used, this log line should be removed and the caller be responsible for logging the error.

If this function stops returning an error and is responsible for logging, `s/glog.V(2).Infof/glog.Errorf/`.